snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -86,7 +86,9 @@ def prepare_columns_and_data_for_insertion(self, data: pd.DataFrame) -> pd.DataF
         data.columns = [self._correct_case(col) for col in original_columns]
         return original_columns, data
 
-    def load_data_into_relation(self, relation: Relation, data: pd.DataFrame) -> None:
+    def load_data_into_relation(
+        self, relation: Relation, data: Optional[pd.DataFrame]
```

Comment:
The previous version of the function accepted empty sets, was it an intentional change?

Reply:
Actually it still is, or am I missing something? If you don't pass anything or pass an empty dataset it will log warning messages, but the function is still gonna proceed and create an empty table. While testing I encountered  some cases where _final sample size_ is 0 and nothing broke during the process.